src/omero_zarr/raw_pixels.py

```diff
@@ -108,88 +74,123 @@ def planeGen() -> np.ndarray:
+    size_c = image.getSizeC()
+    size_z = image.getSizeZ()
+    size_x = image.getSizeX()
+    size_y = image.getSizeY()
+    size_t = image.getSizeT()
+
+    if tile_width is None:
+        tile_width = 1024
+    if tile_height is None:
+        tile_height = 1024
+    tile_width = int(tile_width)
+    tile_height = int(tile_height)
+
+    print(
+        "sizes x: %s, y: %s, z: %s, c: %s, t: %s"
+        % (size_x, size_y, size_z, size_c, size_t)
+    )
+    print(f"tile_width: {tile_width}, tile_height: {tile_height}")
 
-    if cache_dir is not None:
-        cache = True
-    else:
-        cache = False
-        cache_dir = ""
+    chunk_count_x = math.ceil(size_x / tile_width)
+    chunk_count_y = math.ceil(size_y / tile_height)
 
+    # create "0" array if it doesn't exist
+    path = "0"
     dims = [dim for dim in [size_t, size_c, size_z] if dim != 1]
+    shape = tuple(dims + [size_y, size_x])
+    chunks = tuple([1] * len(dims) + [tile_height, tile_width])
+    zarray = parent.require_dataset(
+        path,
+        shape=shape,
+        exact=True,
+        chunks=chunks,
+        dtype=d_type,
+    )
+
+    # Need to be sure that dims match (if array already existed)
+    assert zarray.shape == shape
+    msg = f"Chunks mismatch: existing {zarray.chunks} requested {chunks}"
+    assert zarray.chunks == chunks, msg
 
-    paths: List[str] = []
-    field_groups: List[Array] = []
     for t in range(size_t):
         for c in range(size_c):
             for z in range(size_z):
-                if cache:
-                    assert cache_file_name_func
-                    filename = cache_file_name_func(z, c, t)
-                    if os.path.exists(filename):
-                        plane = numpy.load(filename)
-                    else:
-                        plane = next(planes)
-                        os.makedirs(os.path.dirname(filename), mode=511, exist_ok=True)
-                        numpy.save(filename, plane)
-                else:
-                    plane = next(planes)
-                if plane is None:
-                    continue
-                for level in range(level_count):
-                    size_y = plane.shape[0]
-                    size_x = plane.shape[1]
-                    # If on first plane, create a new group for this resolution level
-                    if len(field_groups) <= level:
-                        path = str(level)
-                        paths.append(path)
-                        field_groups.append(
-                            parent.create(
-                                path,
-                                shape=tuple(dims + [size_y, size_x]),
-                                chunks=tuple([1] * len(dims) + [size_y, size_x]),
-                                dtype=d_type,
-                            )
+                for chk_y in range(chunk_count_y):
+                    for chk_x in range(chunk_count_x):
+                        print(
```

Comment:
I tried the example at https://docs.python.org/3/howto/logging.html#changing-the-format-of-displayed-messages
adding this at the top of `cli.py`

```
import logging
logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.DEBUG)
logging.debug('This message should appear on the console')
logging.info('So should this')
logging.warning('And this, too')
```

but that only printed:
```
WARNING:root:And this, too
```
So it looks like something else is calling `logging.basicConfig` before me, so that my call above is then ignored?

Agreed that we could look to remove `print` statements in a different PR, since this one is already quite long, and `print` is used elsewhere already.